### PR TITLE
Wallet

### PR DESCRIPTION
**Pull request by shweini**

### CODE REVIEW

File: apirouter.go [hunk 1]
```diff
@@ -51,7 +60,6 @@ func newAPIRouter(app *appContext, userRealIP bool) apiMux {
 
 		r.Route("/range/{idx0}/{idx}", func(rd chi.Router) {
 			rd.Use(BlockIndex0PathCtx, BlockIndexPathCtx)
-			rd.Use(middleware.Compress(1))
```
Comment (chappjc) [inline apirouter.go]: Why remove?

File: apirouter.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 
 	"github.com/go-chi/chi"
 	"github.com/go-chi/chi/middleware"
+	//"github.com/dcrdata/dcrdata/wallet"
```
Comment (chappjc) [inline apirouter.go]: remove please

File: apiroutes.go [hunk 3]
```diff
@@ -353,33 +356,21 @@ func (c *appContext) getBlockRangeSummary(w http.ResponseWriter, r *http.Request
...
-		// TODO: deal with the extra newline from Encode, if needed
-		if err := encoder.Encode(c.BlockData.GetSummary(i)); err != nil {
-			apiLog.Infof("JSON encode error: %v", err)
-		}
-		if i != idx {
-			fmt.Fprintf(w, ",%s%s", newline, prefix)
-		}
+		summaries = append(summaries, c.BlockData.GetSummary(i))
 	}
-	fmt.Fprintf(w, "]")
```
Comment (chappjc) [inline apiroutes.go]: looks like this PR is undoing this commit from yesterday:
https://github.com/dcrdata/dcrdata/commit/43540ec3f70ed3f6e7597e12e788cf20405609fc

File: config.go [hunk 4]
```diff
@@ -50,6 +50,8 @@ var (
 	defaultMPTriggerTickets   = 1
 
 	defaultDBFileName = "dcrdata.sqlt.db"
+
+	defaultWalletServer = "127.0.0.1:19110"
```
Comment (chappjc) [inline config.go]: the node default is 9109.  Let's have them both be for mainnet

File: config.go [hunk 5]
```diff
@@ -98,6 +100,8 @@ type config struct {
 	DcrdServ         string `long:"dcrdserv" description:"Hostname/IP and port of dcrd RPC server to connect to (default localhost:9109, testnet: localhost:19109, simnet: localhost:19556)"`
 	DcrdCert         string `long:"dcrdcert" description:"File containing the dcrd certificate file"`
 	DisableDaemonTLS bool   `long:"nodaemontls" description:"Disable TLS for the daemon RPC client -- NOTE: This is only allowed if the RPC client is connecting to localhost"`
+
+	WalletSever      string `long:"walletserver" description:"Hostname/IP and port where dcrwallet is  listening on"`
```
Comment (chappjc) [inline config.go]: Extra space --> "is  listening"

File: config.go [hunk 6]
```diff
@@ -114,6 +118,7 @@ var (
 		MempoolMaxInterval: defaultMempoolMaxInterval,
 		MPTriggerTickets:   defaultMPTriggerTickets,
 		DBFileName:         defaultDBFileName,
+		WalletSever:        defaultWalletServer,
```
Comment (chappjc) [inline config.go]: `WalletSever` --> `WalletServer`

File: config.go [hunk 7]
```diff
@@ -321,6 +326,12 @@ func loadConfig() (*config, error) {
 		cfg.DcrdServ = defaultHost + ":" + activeNet.JSONRPCClientPort
 	}
 
+	//Set the host name and port of the walletserver
+	//if the user does not specify them
+	if cfg.WalletSever == "" {
+		cfg.WalletSever = defaultWalletServer
```
Comment (chappjc) [inline config.go]: `WalletServer`

File: main.go [hunk 8]
```diff
@@ -86,6 +87,14 @@ func mainCore() int {
...
 	}()
 
+
+	//Initialize the dcrwallet funcs for accesing the dcr wallet daemon
+	//This creates its own RPC connection since the wallet daemon
+	//Listens on a different port from the dcrd RPC server
+	dcrwalletClient := wallet.NewWalletClient(cfg.WalletSever, cfg.DcrdUser, cfg.DcrdPass, cfg.DcrdCert, cfg.DisableDaemonTLS)
+
+
+
```
Comment (chappjc) [inline main.go]: Please reduce to single space between neighboring code.

File: main.go [hunk 9]
```diff
@@ -86,6 +87,14 @@ func mainCore() int {
 		time.Sleep(250 * time.Millisecond)
 	}()
 
+
+	//Initialize the dcrwallet funcs for accesing the dcr wallet daemon
+	//This creates its own RPC connection since the wallet daemon
+	//Listens on a different port from the dcrd RPC server
```
Comment (chappjc) [inline main.go]: Please reformat comment `//Initialize` --> `// Initialize`, and join the third line with the second line and wrap as appropriate

File: main.go [hunk 10]
```diff
@@ -328,6 +338,12 @@ func connectNodeRPC(cfg *config) (*dcrrpcclient.Client, semver.Semver, error) {
 		cfg.DcrdCert, cfg.DisableDaemonTLS, notificationHandlers)
 }
 
+
+
+
+
+
+
```
Comment (chappjc) [inline main.go]: Please remove all added new lines here.

File: sample-dcrdata.conf [hunk 11]
```diff
@@ -3,12 +3,16 @@
...
 
-dcrdserv=localhost:9109
+dcrdserv=localhost:19109
 ;dcrdcert=/home/me/.dcrd/rpc.cert
-nodaemontls=0
+nodaemontls=1
+testnet=1
+walletserver=localhost:19110
+
+
```
Comment (chappjc) [inline sample-dcrdata.conf]: Please remove two added new lines

File: wallet/wallet.go [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+import (
+    "fmt"
+    "log"
+    "io/ioutil"
+    "net/http"
+    "encoding/json"
+
+    "github.com/decred/dcrrpcclient"
+)
+type (
```
Comment (chappjc) [inline wallet/wallet.go]: add new line before `type`

File: wallet/wallet.go [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+func NewWalletClient(host, user, pass, cert string, disableTLS bool) *Wallet {
+    return &Wallet{
+        client: connectWalletRPC(host, user, pass, cert, disableTLS),
+    }
+}
+
+func connectWalletRPC(host, user, pass, cert string, disableTLS bool) *dcrrpcclient.Client {
+    ntfnHandlers := dcrrpcclient.NotificationHandlers{}
+
+    fmt.Println(cert)
```
Comment (chappjc) [inline wallet/wallet.go]: for debugging?  please remove if no longer needed

File: config.go (hunk 7 above)
Comment (chappjc) [inline config.go]: The default port would actually depend on if `--testnet` or `--simnet` were used.  I suggest changing the default servers to not include the port, and then append the port for the chosen net.

**Comment by chappjc** [comment]
There needs to be a way to run with only the node (without wallet).

File: config.go (hunk 7 above)
Comment (chappjc) [inline config.go]: Although it would be nice to be able to specify a custom port...

File: config.go (hunk 4 above)
Comment (shweini) [inline config.go]: Wallet daemon listens on port 19110.

**Comment by chappjc** [comment]
Just to clarify, the issue I pointed out was the spelling `WalletSever` -- note the missing "r".  It was fine having `defaultWalletServer` where you had it before.

However, I think this needs PR needs to be done so that the default wallet server is `""`, which signals no wallet connection, *and* there has to be handling everywhere appropriate to deal with this.  dcrdata has to be able to run with just the node as that is an application that needs to be supported.

File: sample-dcrdata.conf [hunk 14]
```diff
@@ -3,12 +3,14 @@
...
+dcrduser=user
+dcrdpass=pass
 
-dcrdserv=localhost:9109
+dcrdserv=localhost:19109
 ;dcrdcert=/home/me/.dcrd/rpc.cert
-nodaemontls=0
+nodaemontls=1
+testnet=1
+walletserver=localhost:19110
```
Comment (raedah) [inline sample-dcrdata.conf]: Id name this dcrwalletserv to keep with the convention above.

File: sample-dcrdata.conf [hunk 15]
```diff
@@ -3,12 +3,14 @@
...
-dcrduser=duser
-dcrdpass=asdfExample
+dcrduser=user
+dcrdpass=pass
 
-dcrdserv=localhost:9109
+dcrdserv=localhost:19109
 ;dcrdcert=/home/me/.dcrd/rpc.cert
-nodaemontls=0
+nodaemontls=1
```
Comment (raedah) [inline sample-dcrdata.conf]: I am unable to get this PR to function with the original nodaemontls=0 setting. 0 should remain the default.

**Comment by chappjc** [comment]
we can find this in the future

**Pull request closed without merging**